crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -131,76 +138,35 @@ impl DiskFileSystem {
             invalidators.into_iter().for_each(|i| i.invalidate());
         }
 
-        watcher_guard.replace(watcher);
+        watcher_guard.replace(debouncer);
 
         spawn_thread(move || {
             let mut batched_invalidate_path = HashSet::new();
             let mut batched_invalidate_path_dir = HashSet::new();
             let mut batched_invalidate_path_and_children = HashSet::new();
             let mut batched_invalidate_path_and_children_dir = HashSet::new();
 
-            'outer: loop {
-                let mut event = rx.recv().map_err(|e| match e {
-                    RecvError => TryRecvError::Disconnected,
-                });
-                loop {
-                    match event {
-                        Ok(DebouncedEvent::Write(path)) => {
-                            batched_invalidate_path.insert(path);
-                        }
-                        Ok(DebouncedEvent::Create(path)) | Ok(DebouncedEvent::Remove(path)) => {
-                            batched_invalidate_path_and_children.insert(path.clone());
-                            batched_invalidate_path_and_children_dir.insert(path.clone());
-                            if let Some(parent) = path.parent() {
-                                batched_invalidate_path_dir.insert(PathBuf::from(parent));
-                            }
-                        }
-                        Ok(DebouncedEvent::Rename(source, destination)) => {
-                            batched_invalidate_path_and_children.insert(source.clone());
-                            if let Some(parent) = source.parent() {
-                                batched_invalidate_path_dir.insert(PathBuf::from(parent));
-                            }
-                            batched_invalidate_path_and_children.insert(destination.clone());
-                            if let Some(parent) = destination.parent() {
+            loop {
+                match rx.recv() {
+                    Ok(Ok(events)) => {
+                        for event in events {
+                            if let Some(parent) = event.path.parent() {
                                 batched_invalidate_path_dir.insert(PathBuf::from(parent));
                             }
+                            batched_invalidate_path_and_children.insert(event.path.clone());
+                            batched_invalidate_path_and_children_dir.insert(event.path);
```

Comment:
The new debouncer implementation has no granularity as to the type of event. `notify` itself now only reports "raw" events. So I'm now doing the same thing for all event types, but I'm not sure whether that's correct.

Reply:
How does these raw events look like?

Reply:
These raw events look promising...

Reply:
But debouncer seems fine too... Let's keep that until we run into problems...